pkg/apis/admissionregistration/types.go

```diff
@@ -458,6 +469,14 @@ type ValidatingAdmissionPolicyBindingSpec struct {
 	//
 	// Required.
 	ValidationActions []ValidationAction
+
+	NamespaceParamRef *NameSpaceParamRef
+}
+
+type NameSpaceParamRef struct {
+	FailAction *ParamFailureActionType
```

Comment:
If I remember right, the semantics of this field were not entirely obvious.  Is is the action that happens if the ref doesn't match anything?

The two cases are?

- Name is non-empty but refers to a non-existent object. This seems like a failure mode, so I agree "FailAction" is a sane name.
- Selector is non-empty, but doesn't select any objects.  This is a special condition. I'd like to support a fallback, but it might be non-obvious to users that this is the condition that triggers a "FailAction".

Reply:
added API docs. Let me know if I you see anything I missed